lib/util/compile.js

```diff
@@ -52,6 +52,14 @@ function which_node_gyp() {
     if (existsSync(node_gyp_bin)) {
         return node_gyp_bin;
     }
+
+    try {
+        var maybe_elsewhere = path.join(path.dirname(fs.realpathSync(which('npm'))), 
```

Comment:
Oops, given the bizarreness of toolchains and installers and nested dips that are needed to trigger the issue, I had to do work inside of an active node_modules and move the work over once I figured it out.  Adding that requirement now.

An alternative to `which` would be baking in the common path that triggers this problem:
`/usr/local/lib/node_modules/npm` (the default place homebrew moves npm when you install node.). The problem is that users can and do change the basepath that homebrew installs too (don't ask me why 😭) or even introduce other npm installs in other parts of their environment.  Fundamentally though, we are looking for `node-gyp` inside of `npm`'s node_modules folder from the users configured environment.  This is the most direct path to find that as far as I can think of.  [`which`](https://www.npmjs.com/package/which) is a fairly tried and true dependency by issacs that npm itself depends on, so its probably a fairly safe bet.

I also suggested that yarn not silent peer-depend on npm and just ship node-gyp like it used when everything 'just worked' to but I'm waiting to hear back about that idea.

Either way, this adds one other fairly robust system check for node-gyp before giving up.